### PR TITLE
schedule, schedulers: refine store state filters

### PR DESCRIPTION
**Pull request by disksing**
## What have you changed? (required)
Currently, there are many store state related schedule filters, including `stateFilter`, `snapshotCountFilter`, `healthFilter`, `disconnectFilter`, `pendingPeerCountFilter`, `blockFilter`, `rejectLeaderFilter`.

Too many types of filters not only lead to verbose code, it is also easy to forget to add a specific filter and causes bugs.

This PR aims to generalize the requirements the merge filters together into a single `StoreStateFilter`.
Complete usage list and update comparison can refer to this document: https://docs.google.com/document/d/1Ks1OougVCjTJA7QfT_ufTfPZoAdrSsWLA6WBkydzWNs/edit?usp=sharing

## What are the type of the changes (required)?
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (required)?
make dev

## Does this PR affect documentation (docs/docs-cn) update? (optional)
No

## Refer to a related PR or issue link (optional)
No

## Benchmark result if necessary (optional)
No

## Add a few positive/negative examples (optional)
No

### CODE REVIEW

File: server/schedule/filters.go [hunk 1]
```diff
@@ -349,3 +349,71 @@ func (f rejectLeaderFilter) FilterSource(opt Options, store *core.StoreInfo) boo
 func (f rejectLeaderFilter) FilterTarget(opt Options, store *core.StoreInfo) bool {
 	return opt.CheckLabelProperty(RejectLeader, store.Labels)
 }
+
+// StoreStateFilter is used to determine whether a store can be selected as the
+// source or target of the schedule based on the store's state.
+type StoreStateFilter struct {
```
Comment (rleungx) [inline server/schedule/filters.go]: Is that better to keep consistent with the other filters?

Comment (disksing) [reply server/schedule/filters.go]: em, how should I change it to 'keep consistent'?

Comment (rleungx) [reply server/schedule/filters.go]: Maybe you can provide something like `NewStoreStateFilter`?

**Comment by liukun4515** [comment]
LGTM

**Pull request merged**